realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -527,55 +528,69 @@ synchronized void invokeWithLock(Callback0 callback) {
      * @throws RealmFileException if copying the file fails.
      */
     private static void copyAssetFileIfNeeded(RealmConfiguration configuration) {
-        IOException exceptionWhenClose = null;
         if (configuration.hasAssetFile()) {
             File realmFile = new File(configuration.getRealmDirectory(), configuration.getRealmFileName());
-            if (realmFile.exists()) {
-                return;
-            }
 
-            InputStream inputStream = null;
-            FileOutputStream outputStream = null;
-            try {
-                inputStream = configuration.getAssetFile();
-                if (inputStream == null) {
-                    throw new RealmFileException(RealmFileException.Kind.ACCESS_ERROR,
-                            "Invalid input stream to asset file.");
-                }
+            copyFileIfNeeded(configuration.getAssetFilePath(), realmFile);
+        }
 
-                outputStream = new FileOutputStream(realmFile);
-                byte[] buf = new byte[4096];
-                int bytesRead;
-                while ((bytesRead = inputStream.read(buf)) > -1) {
-                    outputStream.write(buf, 0, bytesRead);
-                }
-            } catch (IOException e) {
+        // Copy Sync Server certificate path if available
+        String syncServerCertificateAssetName = ObjectServerFacade.getFacade(configuration.isSyncConfiguration()).getSyncServerCertificateAssetName(configuration);
+        if (!Util.isEmptyString(syncServerCertificateAssetName)) {
+            String syncServerCertificateFilePath = ObjectServerFacade.getFacade(configuration.isSyncConfiguration()).getSyncServerCertificateFilePath(configuration);
+
+            File certificateFile = new File(syncServerCertificateFilePath);
+            copyFileIfNeeded(syncServerCertificateAssetName, certificateFile);
```

Comment:
This is a similar use case with the existing [implementation](https://github.com/realm/realm-java/blob/deb25a87d08d93ebc77c5c8d9e81b8a632485b83/realm/realm-library/src/main/java/io/realm/RealmCache.java#L533) i.e what happen if you want to change the Realm shipped in the asset file with the same name. 

For now both use cases will need, the user to uninstall or clear internal data